File: frontend/src/app/features/credit/models/frais-credit-a-encaisser-response.ts
import { StatutDemandeCredit } from './enums/statut-demande-credit.enum';

export interface FraisCreditAEncaisserResponse {
  demandeCreditId: number;
  numeroDemande: string;
  membreNomComplet: string;
  siteNom?: string | null;
  antenneNom?: string | null;
  dateDemande: string;
  montantDemande: number;
  fraisDemande: number;
  fraisDemandePayes: number;
  resteFraisAPayer: number;
  statutDemande: StatutDemandeCredit;
  objetCredit?: string | null;
  sessionCaisseRequise: boolean;
}

File: backend/src/main/java/com/mini/credit/dto/credit/FraisCreditAEncaisserResponse.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.StatutDemandeCredit;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Builder
public class FraisCreditAEncaisserResponse {
    private Long demandeCreditId;
    private String numeroDemande;
    private String membreNomComplet;
    private String siteNom;
    private String antenneNom;
    private LocalDate dateDemande;
    private BigDecimal montantDemande;
    private BigDecimal fraisDemande;
    private BigDecimal fraisDemandePayes;
    private BigDecimal resteFraisAPayer;
    private StatutDemandeCredit statutDemande;
    private String objetCredit;
    private boolean sessionCaisseRequise;
}